~[T] has been removed from rust-master, causing a build failure

This is the error, for reference:

(./configure && make)

``` bash
configure: no SSL library selected or found
configure: you will not be able to access HTTPS URLs
configure: check the README for instructions on enabling HTTPS
configure: writing Makefile
configure: done!
mkdir -p build/
rustc src/codegen/main.rs --out-dir=build
src/codegen/status.rs:27:1: 29:2 warning: function `StatusC` should have a snake case name such as `status_c`, #[warn(non_snake_case_functions)] on by default
src/codegen/status.rs:27 fn StatusC(code: uint, reason: &'static str, comment: &'static str) -> HeadingOrStatus {
src/codegen/status.rs:28     Status(Status { code: code, reason: reason, comment: Some(comment) })
src/codegen/status.rs:29 }
src/codegen/status.rs:32:1: 34:2 warning: function `StatusN` should have a snake case name such as `status_n`, #[warn(non_snake_case_functions)] on by default
src/codegen/status.rs:32 fn StatusN(code: uint, reason: &'static str) -> HeadingOrStatus {
src/codegen/status.rs:33     Status(Status { code: code, reason: reason, comment: None })
src/codegen/status.rs:34 }
mkdir -p src/http/generated
build/codegen read_method.rs src/http/generated/
build/codegen status.rs src/http/generated/
mkdir -p build/
rustc -O  src/http/lib.rs --out-dir=build
src/http/memstream.rs:96:25: 96:26 error: obsolete syntax: `~[T]` is no longer a type
src/http/memstream.rs:96         let mut buf = ~[];
                                                 ^
note: use the `Vec` type instead
src/http/memstream.rs:99:26: 99:27 error: obsolete syntax: `~[T]` is no longer a type
src/http/memstream.rs:99         let mut buf = ~[0];
                                                  ^
src/http/memstream.rs:103:35: 103:36 error: obsolete syntax: `~[T]` is no longer a type
src/http/memstream.rs:103         let mut buf = ~[0, 0, 0, 0];
                                                            ^
error: aborting due to 3 previous errors
make: *** [build/.libhttp.timestamp] Error 101
```

--- a/src/http/memstream.rs
+++ b/src/http/memstream.rs
@@ -93,17 +93,17 @@ mod test {
     #[test]
     fn test_mem_reader_fake_stream() {
         let mut reader = MemReaderFakeStream::new(vec!(0, 1, 2, 3, 4, 5, 6, 7));
-        let mut buf = ~[];
+        let mut buf = vec![];
         assert_eq!(reader.read(buf),      Ok(0));
         assert_eq!(reader.tell(),         Ok(0));
-        let mut buf = ~[0];
+        let mut buf = vec![0];
         assert_eq!(reader.read(buf),      Ok(1));
         assert_eq!(reader.tell(),         Ok(1));
-        assert_eq!(buf,                   ~[0]);
-        let mut buf = ~[0, 0, 0, 0];
+        assert_eq!(buf,                   vec![0]);
+        let mut buf = vec![0, 0, 0, 0];
         assert_eq!(reader.read(buf),      Ok(4));
         assert_eq!(reader.tell(),         Ok(5));
-        assert_eq!(buf,                   ~[1, 2, 3, 4]);
+        assert_eq!(buf,                   vec[1, 2, 3, 4]);
         assert_eq!(reader.read(buf),      Ok(3));
         assert_eq!(buf.slice(0, 3),       &[5, 6, 7]);
         assert_eq!(reader.read(buf).ok(), None);

